### PR TITLE
BUG: Fixed __join_multi always returning indexers (#34074)

### LINKED ISSUE #34074: BUG: Multijoining indexes always returns indexers, ignores function parameters
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample

```python
>>>frame = pd.DataFrame([(1,2,3,4),(4,5,6,7)],columns=["i1", "i2", "i3","val"]).set_index(["i1","i2","i3"])
>>>frame2 = pd.DataFrame([(1,2,3),(4,5,6)],columns=["i1", "i2", "val"]).set_index(["i1","i2"])
>>>frame.index.join(frame2.index,return_indexers=False)
(MultiIndex([(1, 2, 3),
            (4, 5, 6)],
           names=['i1', 'i2', 'i3']), None, None)
```

#### Problem description
Parameter return_indexers is ignored in some cases of __join_multi

#### Expected Output
```python
>>>frame.index.join(frame2.index,return_indexers=False)
MultiIndex([(1, 2, 3),
            (4, 5, 6)],
           names=['i1', 'i2', 'i3'])
```

### PR DESCRIPTION
**Pull request by CuylenE**
- [x] closes #34074
- [x] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks @CuylenE 

Is there a test case for the `return_indexers=False` case? If not, can you add one?

File: pandas/tests/indexes/multi/test_join.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def test_join_multi_wrong_order():
     midx1 = pd.MultiIndex.from_product([[1, 2], [3, 4]], names=["a", "b"])
     midx2 = pd.MultiIndex.from_product([[1, 2], [3, 4]], names=["b", "a"])
 
-    join_idx, lidx, ridx = midx1.join(midx2, return_indexers=False)
+    join_idx, lidx, ridx = midx1.join(midx2, return_indexers=True)
```
Comment (jreback) [inline pandas/tests/indexes/multi/test_join.py:99]: can you add a test that hits return=False

**Review by jreback — CHANGES_REQUESTED** [review]
also pls add a whatsnew note, bug fixes 1.1., MultiIndex section

File: pandas/tests/indexes/multi/test_join.py [hunk 2]
```diff
@@ -96,10 +96,19 @@ def test_join_multi_wrong_order():
...
     tm.assert_numpy_array_equal(ridx, exp_ridx)
+
+
+def test_join_multi_return_indexers():
+    # GH 34074
+
+    midx1 = pd.MultiIndex.from_product([[1, 2], [3, 4], [5, 6]], names=["a", "b", "c"])
+    midx2 = pd.MultiIndex.from_product([[1, 2], [3, 4]], names=["a", "b"])
+
+    assert type(midx1.join(midx2, return_indexers=False)) != tuple
```
Comment (MarcoGorelli) [inline pandas/tests/indexes/multi/test_join.py:114]: Could this be made more explicit?

Just a suggestion:
```python
result = midx1.join(midx2, return_indexers=False)
expected = <construct expected result here>
tm.assert_index_equal(result, expected)
```

As an aside, see [PEP8](https://www.python.org/dev/peps/pep-0008/) for how it's advised to write object type comparisons

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks @CuylenE , just making a suggestion for the test

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @CuylenE

**Pull request merged**